routes/web.php

```diff
@@ -0,0 +1,19 @@
+<?php
+
+use Illuminate\Support\Facades\Route;
+use DrewRoberts\Blog\Http\Controllers\PageController;
+use DrewRoberts\Blog\Http\Controllers\TopicController;
+use DrewRoberts\Blog\Http\Middleware\ResolvePage;
+
+
+Route::middleware(config('tipoff.web.middleware_group'))
+    ->prefix(config('tipoff.web.uri_prefix'))
+    ->group(function () {
+
+        Route::get('{page}/{child-page}/{grandchild-page}', PageController::class)
+            ->name('page');
+
+            Route::get('{topic}/{series}/{post}', TopicController::class)
+            ->name('blog');
```

Comment:
Neither of these routes will work - they will match against ALL 3 segment paths.  To be distinct, both routes need some fixed segment in the path to match against.  Ex `/page/{page}` and `/blog/{topic}/{series}/{post}`

Also, I have no idea what {child-page} and {grandchild-page} are or why they are required.  If they are optional, they either need to be flagged as optional or, more likely you'll need multiple routes with different numbers of segments.   You'll also need to register explicit bindings between the route pattern and the model and make sure the controller parameters are named properly.

For the `/blog/{topic}/{series}/{post}` route - are series and post always required?  If not, you probably want to register 3 routes - each with a different controller.

```
Route::get('/blog/{topic}', TopicController::class)->name('topic');
Route::get('/blog/{topic}/{series}', SeriesController::class)->name('series');
Route::get('/blog/{topic}/{series}/{post}', PostController::class)->name('post');
```